table/evaluators.go

```diff
@@ -0,0 +1,1125 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package table
+
+import (
+	"fmt"
+	"math"
+	"slices"
+
+	"github.com/apache/iceberg-go"
+	"github.com/google/uuid"
+)
+
+const (
+	rowsMightMatch, rowsMustMatch      = true, true
+	rowsCannotMatch, rowsMightNotMatch = false, false
+	inPredicateLimit                   = 200
+)
+
+// newManifestEvaluator returns a function that can be used to evaluate whether a particular
+// manifest file has rows that might or might not match a given partition filter by using
+// the stats provided in the partitions (UpperBound/LowerBound/ContainsNull/ContainsNaN).
+func newManifestEvaluator(spec iceberg.PartitionSpec, schema *iceberg.Schema, partitionFilter iceberg.BooleanExpression, caseSensitive bool) (func(iceberg.ManifestFile) (bool, error), error) {
+	partType := spec.PartitionType(schema)
+	partSchema := iceberg.NewSchema(0, partType.FieldList...)
+	filter, err := iceberg.RewriteNotExpr(partitionFilter)
+	if err != nil {
+		return nil, err
+	}
+
+	boundFilter, err := iceberg.BindExpr(partSchema, filter, caseSensitive)
+	if err != nil {
+		return nil, err
+	}
+
+	return (&manifestEvalVisitor{partitionFilter: boundFilter}).Eval, nil
+}
+
+type manifestEvalVisitor struct {
```

Comment:
I'm a little confused, is this new code or just code that has been moved around?

Reply:
This is just code that was moved around. I was able to put it into the table package to limit access. This doesn't need to be exported so I was able to shift this into the table package rather than export it in the main iceberg package.